File: src/main/java/gestorAplicacion/comunicacion/ServicioGenerico.java
package gestorAplicacion.comunicacion;

/**
 * Implementación de un servicio de envío específico usando todos los medios.
 *
 * @author Juan Daniel Bula, Leidy Daniela Alzate, Oscar Andres Usuga
 */
public class ServicioGenerico extends ServicioEnvio {

    /**
     * Constructor asignando el mensaje a todos los medios.
     * @param mensaje mensaje a ser enviado.
     */
    public ServicioGenerico(String mensaje) {
        super(mensaje);
    }

    @Override
    public void enviar() {
        new ServicioEmail(mensaje).enviar();
        new ServicioSms(mensaje).enviar();
        new ServicioFax(mensaje).enviar();
    }
}

File: python/gestorAplicacion/comunicacion/servicio_generico.py
from gestoraplicacion.comunicacion.servicio_email import ServicioEmail
from gestoraplicacion.comunicacion.servicio_envio import ServicioEnvio
from gestoraplicacion.comunicacion.servicio_fax import ServicioFax
from gestoraplicacion.comunicacion.servicio_sms import ServicioSms


class ServicioGenerico(ServicioEnvio):
    """
     Constructor asignando el mensaje a todos los medios.
     @param mensaje mensaje a ser enviado.
    """
    def __init__(self, mensaje):
        super().__init__(mensaje)

    def enviar(self):
        ServicioEmail(super()._mensaje).enviar()
        ServicioSms(super()._mensaje).enviar()
        ServicioFax(super()._mensaje).enviar()
